### ISSUE TITLE
Problem with encoding, needs some info to understand

### ISSUE THREAD

**Issue by ciaoben** [net-ssh/net-ssh#392]
```
def yell command, sudo= ""
    Net::SSH.start(remote_ip, username) do |ssh|
      ssh.open_channel do |channel|
        channel.exec("#{sudo}#{command}") do |ch, success|
          unless success
            raise RuntimeError, "FAILED SSH Exec: couldn't execute #{command}"
          end
          channel.on_data do |ch, data|
            stdout_data+=data
          end
          channel.on_extended_data do |ch, type, data|
            stderr_data+=data
          end
          channel.on_request("exit-status") do |ch, data|
            exit_code = data.read_long
          end
          channel.on_request("exit-signal") do |ch, data|
            exit_signal = data.read_long
          end
        end
      end
    end
end
```

given this basic function that use this library, it happens something I can really understand:

```
command = 'echo "eeeeeeeeeeeeeeeeeeeeeeeeeewwowowowìeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee"'

yell command
```

the command "echo ..." is 126 char long.

it returns me what I want:
```
["eeeeeeeeeeeeeeeeeeeeeeeeeewwowowow\xC3\xACeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee\n",
 "",
 0,
 nil]
```

(except for the `\xC3\xAC`)

if I call the methods with the same command but with 1 more char (total length 127):

```
 command = 'echo "eeeeeeeeeeeeeeeeeeeeeeeeeewwowowowìeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee"'
yell command
```

it return me this:

```
Encoding::CompatibilityError: incompatible character encodings: ASCII-8BIT and UTF-8
from /usr/local/lib/ruby/gems/2.3.0/gems/net-ssh-3.0.2/lib/net/ssh/buffer.rb:284:in `block in write'
```

and if I call a 127 long string, without the special char `ì`:

```
 command = 'echo "eeeeeeeeeeeeeeeeeeeeeeeeeewwowowoweeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee"'
yell command
```

it returns me again what I want:
```
["eeeeeeeeeeeeeeeeeeeeeeeeeewwowowow\xC3\xACeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee\n",
 "",
 0,
 nil]
```

I can't understand why this happen, what am I missing?

**Comment by mfazekas** [comment]
There is a bug in net-ssh buffer. It contains raw binary data as well as utf-8 string which can cause issues. Please see PR #394.

**Comment by mfazekas** [comment]
should be fixed in 4.0.0.beta1
